internal/query/roles_v2_test.go

```diff
@@ -0,0 +1,392 @@
+package query
+
+import (
+	"context"
+	"testing"
+
+	pb "github.com/authzed/authzed-go/proto/authzed/api/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.infratographer.com/permissions-api/internal/iapl"
+	"go.infratographer.com/permissions-api/internal/testingx"
+	"go.infratographer.com/permissions-api/internal/types"
+)
+
+func rbacv2TestPolicy() iapl.Policy {
+	p := DefaultPolicyV2()
+
+	if err := p.Validate(); err != nil {
+		panic(err)
+	}
+
+	return p
+}
+
+func rbacV2CreateParentRel(parent, child types.Resource, namespace string) []*pb.RelationshipUpdate {
+	return []*pb.RelationshipUpdate{
+		{
+			Operation: pb.RelationshipUpdate_OPERATION_TOUCH,
+			Relationship: &pb.Relationship{
+				Resource: resourceToSpiceDBRef(namespace, child),
+				Relation: "parent",
+				Subject: &pb.SubjectReference{
+					Object: resourceToSpiceDBRef(namespace, parent),
+				},
+			},
+		},
+		{
+			Operation: pb.RelationshipUpdate_OPERATION_TOUCH,
+			Relationship: &pb.Relationship{
+				Resource: resourceToSpiceDBRef(namespace, child),
+				Relation: "parent",
+				Subject: &pb.SubjectReference{
+					Object:           resourceToSpiceDBRef(namespace, parent),
+					OptionalRelation: "parent",
+				},
+			},
+		},
+		{
+			Operation: pb.RelationshipUpdate_OPERATION_TOUCH,
+			Relationship: &pb.Relationship{
+				Resource: resourceToSpiceDBRef(namespace, parent),
+				Relation: "member",
+				Subject: &pb.SubjectReference{
+					Object:           resourceToSpiceDBRef(namespace, child),
+					OptionalRelation: "member",
+				},
+			},
+		},
+	}
+}
+
+func TestCreateRolesV2(t *testing.T) {
+	namespace := "testroles"
+	ctx := context.Background()
+	e := testEngine(ctx, t, namespace, rbacv2TestPolicy())
+
+	tenant, err := e.NewResourceFromIDString("tnntten-root")
+	require.NoError(t, err)
+	actor, err := e.NewResourceFromIDString("idntusr-actor")
+	require.NoError(t, err)
+
+	// group is not a role owner in this policy
+	invalidOwner, err := e.NewResourceFromIDString("idntgrp-group")
+	require.NoError(t, err)
+
+	type input struct {
+		name    string
+		actions []string
+		owner   types.Resource
+	}
+
+	tc := []testingx.TestCase[input, []types.Role]{
+		{
+			Name: "InvalidActions",
+			Input: input{
+				name:    "role1",
+				actions: []string{"action1", "action2"},
+				owner:   tenant,
+			},
+			CheckFn: func(ctx context.Context, t *testing.T, res testingx.TestResult[[]types.Role]) {
+				require.Error(t, res.Err)
+				assert.Len(t, res.Success, 0)
+			},
+		},
+		{
+			Name: "InvalidOwner",
+			Input: input{
+				name:  "lb_viewer",
+				owner: invalidOwner,
+				actions: []string{
+					"loadbalancer_list",
+					"loadbalancer_get",
+				},
+			},
+			CheckFn: func(ctx context.Context, t *testing.T, res testingx.TestResult[[]types.Role]) {
+				require.Error(t, res.Err)
+				assert.ErrorContains(t, res.Err, "not allowed on relation")
```

Comment:
You should be able to use `ErrorIs` instead of `ErrorContains`. Same goes for other uses elsewhere.